cypress/support/commands.js

```diff
@@ -1,3 +1,31 @@
-// ***********************************************
-// Create various custom commands and overwrite existing commands.
-// ***********************************************
+Cypress.Commands.add(
```

Comment:
This feels like an API tests, not UI. Isn't it possible to click over the frontend to do that?

Reply:
It's not a test. I added it as a utility to be used in other tests where we're not explicitly testing for resource creation.
To test app deployment/creation from the UI, we can add a test for that particular aspect, where we click over the UI to do it, but other cases might be able to use this utility method to test only features that they're concerned with.

Reply:
Still, UI e2e are supposed to do their tests by clicking over the UI. If page objects will be created it will be much easier to write this. Also, it's best to write e2e in the form of stories. 

1. Going to deploy from file
2. Deploying crd
3. Going to crd list view
4. Should contain x
5. Delete crd
6. Should show empty list
etc.

This way the test will simulate user actions otherwise it's not much different than unit tests with mocked API responses.

Reply:
I've tried to rewrite the tests in form of stories.

I also tried to create resources via the input form at `/#/create`, but I couldn't find a good approach.

Using `cypress.type` inputs wrongly-formatted YAML because pressing `enter` on Ace Editor preserves indent, and the YAML file has its own indent so we get something like:
![image](https://user-images.githubusercontent.com/11166933/63802264-1f177e00-c90a-11e9-8aec-93a4ff5e8627.png)

Setting the value automatically via the Ace module, (e.g. via below snippet) does not trigger Angular's change detection so the input is not formatted, and the `Upload` button is greyed out.

Snippet:
```js
import * as ace from 'brace';
...
cy.get('ace-editor').then(ele => {
  const editor = ace.edit(ele[0]);
  editor.setValue(content);
  editor.clearSelection();
});
```
Display:
![image](https://user-images.githubusercontent.com/11166933/63802533-c09ecf80-c90a-11e9-95da-fbfcda2b940d.png)

Reply:
How about using `setValue` + typing new line to make it run the validation code?

Reply:
I tried that, didn't work. The script sees the entire input value as a newline (`\n`) (or whatever else I typed).